### PR TITLE
feat(reorderSiteAwards): make some minor improvements to desktop ux

### PR DESCRIPTION
**Pull request by wescopeland**
This PR makes a few minor adjustments to the reorder site awards UX. The changes are most impactful on desktop.

**Summary of Changes**
* A minor bug has been fixed where manual move buttons were hidden when the "Hidden" checkbox was ticked, but not truly disabled.
* The rows themselves are now more compact, allowing a user to see more of them (and thus, drag and drop farther) on the screen.
* The page now auto-scrolls when the user is dragging and holds their cursor near the top or bottom edges of the window.
* A new "Hide saved hidden rows" checkbox has been added to the top of the page. When checked, saved hidden rows will be given a `hidden` class, rendering them invisible.

From a technical implementation standpoint, I figured it also made sense to start separating out the single _reorderSiteAwards.ts_ file into multiple small utils. **Almost everywhere I have changed something, I have done this. As a result of moving the original file, the adds/deletes on this PR are much larger than the actual code that has been contributed and removed.** This will make future enhancements I have planned a bit easier to write. I would also like to add tests, and those will be easier to develop once everything is broken out.

**Awards Size Before**
Awards Size Old

**Awards Size After**
Awards Size New

---

https://user-images.githubusercontent.com/3984985/235556630-610f29a9-7482-4c1a-bebe-d675dcc399c4.mp4

### CODE REVIEW

**Review by Jamiras — COMMENTED** [review]
Not sure if I missed a post-checkout command, but none of the reordering buttons seem to do anything for me. Drag/drop and hide do work.

File: public/reorderSiteAwards.php [hunk 1]
```diff
@@ -281,8 +309,20 @@ function RenderAwardOrderTable(string $title, array $awards, int &$awardCounter,
...
-                    <button onclick='handleSaveAllClick()' class='mt-2 mb-6 text-base'>Save All Changes</button>
+                <div class="flex w-full items-center justify-between mt-3 mb-6">
+                    <div class="flex items-center gap-x-1">
+                        <label class="flex items-center gap-x-1">
+                            <input
+                                type="checkbox"
+                                onchange="reorderSiteAwards.handleHideSavedHiddenRowsChange(event)"
+                                $hideSavedHiddenRowsCheckedAttribute
+                            >
+                                Hide saved hidden rows
```
[the remark this answers is not in the document]
Comment (wescopeland) [reply public/reorderSiteAwards.php]: > Not sure if I missed a post-checkout command, but none of the reordering buttons seem to do anything for me. Drag/drop and hide do work.

You didn't; I had a bug. Should be fixed now.

I've also changed the wording of the checkbox (and its behavior to match). It is still unchecked by default, and the setting still persists albeit with a new cookie name.

[the remark this answers is not in the document]
Comment (Jamiras) [reply public/reorderSiteAwards.php]: Minor: with the checkbox unchecked, the rows for newly hidden items aren't made invisible after saving.

[the remark this answers is not in the document]
Comment (wescopeland) [reply public/reorderSiteAwards.php]: > Minor: with the checkbox unchecked, the rows for newly hidden items aren't made invisible after saving.

Digging into this now, I'd like to get this fixed before approval.

[the remark this answers is not in the document]
Comment (wescopeland) [reply public/reorderSiteAwards.php]: I think this should now be resolved. I'm testing around with the checkbox ticked and unticked, things seem okay. Let me know if you see any more weirdness and I will tackle it.

**Review by Jamiras — APPROVED** [review]

**Pull request merged**